asyncauth.go

```diff
@@ -191,27 +199,31 @@ func (config *Config) renderInstructions(w http.ResponseWriter, req *http.Reques
 	}
 
 	selectCluster := req.URL.Query().Get("cluster")
-	var cluster Cluster
+	var cluster *Cluster
 	if selectCluster == "" {
-		cluster = config.Clusters[0]
+		cluster = config.getFirstClusterOrPanic()
 	} else {
-		found := false
 		for _, c := range config.Clusters {
 			parsed, _ := url.Parse(c.K8s_Master_URI)
 			if parsed.Hostname() == selectCluster {
-				cluster = c
-				found = true
+				cluster = &c
```

Comment:
nit: `cluster` refers to the value of `c` in the last iteration. It's been a (subtle) source of bugs for me, and for that reason prefer to reference the original value (using the index), i.e., `cluster = &config.Clusters[i]`. In this case, there's no problem because of the `break`.

Reply:
:+1: